app/models/identity.rb

```diff
@@ -34,8 +34,12 @@ def self.find_or_create_from_oauth(auth_hash)
                               :token   => auth_hash[:credentials][:token],
                               :service => auth_hash[:provider],
                               :name    => "#{provider} - #{auth_hash[:info][:name]}")
+      identity.refresh_token = auth_hash[:credentials][:refresh_token] unless auth_hash[:credentials][:refresh_token].nil?
+      identity.expires_at = auth_hash[:credentials][:expires_at] unless auth_hash[:credentials][:expires_at].nil?
     else
       identity.token = auth_hash[:credentials][:token]
+      identity.refresh_token = auth_hash[:credentials][:refresh_token] unless auth_hash[:credentials][:refresh_token].nil?
```

Comment:
In order to shorten these lines, perhaps a block can be used instead of this inline `unless`? I'ts pretty hard to read now.